org.eclipse.scanning.sequencer/src/org/eclipse/scanning/sequencer/nexus/NexusScanFileManager.java

```diff
@@ -353,19 +350,29 @@ protected int getScanRank(Iterable<IPosition> gen) {
 		return devices.stream().map(d -> d.getName()).collect(Collectors.toSet());
 	}
 	
-	protected ScanPointsWriter createScanPointsWriter() {
-		ScanPointsWriter scanPointsWriter = new ScanPointsWriter(model);
+	protected SolsticeScanMonitor createSolsticeScanMonitor(ScanModel scanModel) {
+		SolsticeScanMonitor scanPointsMonitor = new SolsticeScanMonitor(model);
 		
 		// get the nexus object providers for all device types excluding metadata scannables
 		EnumSet<ScanRole> deviceTypes = EnumSet.complementOf(EnumSet.of(ScanRole.METADATA));
-		List<NexusObjectProvider<?>> nexusObjectProviders = this.nexusObjectProviders.entrySet().stream()
+		List<NexusObjectProvider<?>> nexusObjectProviderList = nexusObjectProviders.entrySet().stream()
 			.filter(e -> deviceTypes.contains(e.getKey())) // filter where key is in deviceType set
 			.flatMap(e -> e.getValue().stream())  // concatenate value lists into a single stream
 			.collect(Collectors.toList());  // collect in a list
 		
-		scanPointsWriter.setNexusObjectProviders(nexusObjectProviders);
+		scanPointsMonitor.setNexusObjectProviders(nexusObjectProviderList);
 		
-		return scanPointsWriter;
+		// add the solstice scan monitor to the list of monitors in the scan
+		List<IScannable<?>> monitors = scanModel.getMonitors();
+		if (monitors == null) {
+			monitors = new ArrayList<>();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "monitors". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1854)